qa/common/common.sh

```diff
@@ -726,3 +726,29 @@ echo "Result: OK"
 EOF
     _run_test_script_on_node $TESTSCRIPT $STORAGENODE
 }
+
+function configure_OSD_to_filestore {
+	salt-run proposal.populate format=filestore name=filestore 
+	chown salt:salt /srv/pillar/ceph/proposals/policy.cfg
+	sed -i 's/profile-default/profile-filestore/g' /srv/pillar/ceph/proposals/policy.cfg
+}
+
+function check_OSD_type {
+	salt \* pillar.get ceph
+	salt \* cmd.run "for i in \$(mount|grep ceph|awk '{print \$3}');do cat \${i}/type;done"
```

Comment:
This function does not verity the expected OSD_TYPE. That effectively means that you would need to inspect the logs in order to find out if the test passed or not.

Adding a 'expected' param which can be set to either filestore or bluestore will help.

Reply:
note that salt returns 0 regardless of whether the remote command succeeded or not, so you have to grep the results. In this case, it's a simple case of grepping for any instances of the "other" possibility. For example, if all the OSDs are expected to be filestore, grep the output for bluestore. And vice versa.

The mere absence of a string in the output is not sufficient as success criteria, however. I would also count the number of instances of the string that is expected and check that it is the same as the number of OSDs in the cluster. (There is a function `json_total_osds` for determining this.)